assert/assertions.go

```diff
@@ -1336,7 +1336,15 @@ func PanicsWithError(t TestingT, errString string, f PanicTestFunc, msgAndArgs .
 	}
 	panicErr, ok := panicValue.(error)
 	if !ok || panicErr.Error() != errString {
-		return Fail(t, fmt.Sprintf("func %#v should panic with error message:\t%#v\n\tPanic value:\t%#v\n\tPanic stack:\t%s", f, errString, panicValue, panickedStack), msgAndArgs...)
+		errorMsg := ""
+		if ok {
+			errorMsg = fmt.Sprintf("\tError message:\t%#v\n", panicErr.Error())
+		}
+		msg := fmt.Sprintf(
+			"func %#v should panic with error message:\t%#v\n\tPanic value:\t%#v\n%s\tPanic stack:\t%s",
+			f, errString, panicValue, errorMsg, panickedStack,
+		)
```

Comment:
There is a cleaner implementation here that doesn't require the errorMsg variable. Either using strings.Builder or by appending to msg we can remove errorMsg and significantly clean up the Sprintf format string, which was already quite messy.